x-pack/plugins/fleet/server/services/epm/kibana/assets/tag_assets.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { taggableTypes } from '@kbn/saved-objects-tagging-plugin/common/constants';
+import type { IAssignmentService, ITagsClient } from '@kbn/saved-objects-tagging-plugin/server';
+
+import type { KibanaAssetType } from '../../../../../common';
+
+import type { ArchiveAsset } from './install';
+
+const TAG_COLOR = '#FFFFFF';
+const MANAGED_TAG_NAME = 'Managed';
+
+export async function tagKibanaAssets({
+  savedObjectTagAssignmentService,
+  savedObjectTagClient,
+  kibanaAssets,
+  pkgTitle,
+}: {
+  savedObjectTagAssignmentService: IAssignmentService;
+  savedObjectTagClient: ITagsClient;
+  kibanaAssets: Record<KibanaAssetType, ArchiveAsset[]>;
+  pkgTitle: string;
+}) {
+  const taggableAssets = Object.entries(kibanaAssets).flatMap(([assetType, assets]) => {
+    if (!taggableTypes.includes(assetType as KibanaAssetType)) {
+      return [];
+    }
+
+    if (!assets.length) {
+      return [];
+    }
+
+    return assets;
+  });
+
+  const allTags = await savedObjectTagClient.getAll();
```

Comment:
For some reason `savedObjectTagClient` is `undefined` in `fleet_api_integration` test using FTR.
Any idea how to fix this?
```
  │ proc [kibana] [2022-07-26T15:58:45.944+00:00][ERROR][plugins.fleet] TypeError: Cannot read properties of undefined (reading 'getAll')
[00:00:52]           │ proc [kibana]     at tagKibanaAssets (/var/lib/buildkite-agent/builds/kb-n2-4-spot-abdcfdd63a96450d/elastic/kibana-pull-request/kibana-build-xpack/x-pack/plugins/fleet/server/services/epm/kibana/assets/install.js:186:46)
[00:00:52]           │ proc [kibana]     at installKibanaAssetsAndReferences (/var/lib/buildkite-agent/builds/kb-n2-4-spot-abdcfdd63a96450d/elastic/kibana-pull-request/kibana-build-xpack/x-pack/plugins/fleet/server/services/epm/kibana/assets/install.js:157:9)
[00:00:52]           │ proc [kibana]     at runMicrotasks (<anonymous>)
[00:00:52]           │ proc [kibana]     at processTicksAndRejections (node:internal/process/task_queues:96:5)
[00:00:52]           └- ✖ fail: Fleet Endpoints EPM Endpoints EPM - get returns correct package info if it was installed by upload
[00:00:52]           │      Error: expected 200 "OK", got 500 "Internal Server Error"
```

Reply:
What I'm finding very odd about this is that the `Saved object tagging start service not set.` error isn't being thrown from the app context service. This seems to me to indicate that the `createTagClient` is actually returning undefined. Is there some kind of mock of the tagging plugin or our app context service?

Reply:
I don't see any mocking of these services, we are calling the API in the integration test.

Reply:
I only found one usage of `appContextService` in api tests, it looks like it was called manually for some reason in template test:
https://github.com/elastic/kibana/blob/main/x-pack/test/fleet_api_integration/apis/epm/template.ts#L21

Reply:
I found the issue, it was a coding error, I didn't notice there are two invocations of `_installPackage`, and didn't pass the parameters from `installPackageByUpload`